Migrate THEOlive docs (#202)
THEOplayer/documentation: theolive/getting-started/authorization.mdx

---
sidebar_position: 1

> sidebar_label: Authorization

Reviewer: Remove `sidebar_label`, `pagination_label` and `title` from all pages, and use a Markdown level 1 heading instead to specify the title. For example:
```markdown
# Authorization

We make sure that only authorized users can make API calls!

---

```

pagination_label: Authorization

> pagination_next: getting-started/regions

Reviewer: You should only need to set `pagination_next` on `index` pages, for all other pages Docusaurus should already paginate correctly. Could you try removing these and see if it still works?

Author: Yep, still works. Thank you :)

title: Autorization
---

We make sure that only authorized users can make API calls!

---

Making secure calls to our REST API requires authentication and authorization. THEOlive makes use of token-secret pairs to authorize requests.

## 1. Getting a token-secret pair

To generate a token-secret pair, you should be registered for an account at on our [management console](https://console.theo.live/). Under the section "Tokens", click on the "Generate Token" button, and enter a name before clicking "Generate".

![](../assets/img/c0aca77-1.-THEOlive-console-generate-token.PNG)

This will give a pop-up which shows your token "key" and the corresponding secret.

![](../assets/img/bb9b4a6-2.-THEOlive-console-generated-token-and-secret.PNG)

Once you "Close" the pop-up window, the new token will be visible in your list of tokens.

![](../assets/img/5361242-3.-THEOlive-console-token-list.PNG)

> > 🚧 Please store the secret
> >
> > Make sure to store the secret somewhere, as this will be shown to you only once, and THEOlive only stores a hash of the secret, so it cannot be recovered afterward.

Reviewer: Pro tip: try using [admonitions](https://docusaurus.io/docs/markdown-features/admonitions) to make these prettier. 😁 

For example:
```suggestion
:::warning[Please store the secret]

Make sure to store the secret somewhere, as this will be shown to you only once, and THEOlive only stores a hash of the secret, so it cannot be recovered afterward.

:::
```

In case you forgot your secret, or the secret got exposed, you can easily regenerate a pair at all times.

## 2. Using the token-secret pair with Basic Authentication

> THEOlive makes use of Basic Authentication when making requests to the API. In the 'Authorization' header, make sure to pass your token-secret pair (encrypted with base64), prepended with the word 'Basic'.

Reviewer: Nit-pick: base64 is not encryption. 😛

```suggestion
THEOlive makes use of Basic Authentication when making requests to the API. In the 'Authorization' header, make sure to pass your token-secret pair (encoded with base64), prepended with the word 'Basic'.
```

As an example, assume that your token has the value "my-token", and your secret has the value "my-secret". Then you can put them together like this: `my-token:my-secret`.

> You subsequently encode this value using base64. Applying 'base64(my-token:my-secret)' will result in the following encoded string: 'bXktdG9rZW46bXktc2VjcmV0'

Reviewer: ```suggestion
You subsequently encode this value using base64. Applying `base64(my-token:my-secret)` will result in the following encoded string: `bXktdG9rZW46bXktc2VjcmV0`
```

---
sidebar_position: 1
---

> # Autorization

Reviewer: Small typo:
```suggestion
# Authorization
```
